src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,124 @@
+    }
+
+    public MyHashMap() {
+        table = (Node<K, V>[])new Node[INITIAL_CAPACITY];
+        threshold = (int)(table.length * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (++size > threshold) {
+            table = resize();
+        }
+        putToNewTable(new Node<K, V>(key, value), table);
+        threshold = (int)(table.length * LOAD_FACTOR);
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key, table.length);
+        if (table[index] != null) {
+            Node<K, V> foundNode = searchInBacket(key, index);
+            return foundNode.value;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private Node<K,V> searchInBacket(K key, int index) {
+        Node<K,V> current = table[index];
+        while (current != null) {
+            if (current.key == key
+                    || key != null && key.equals(current.key)) {
+                return current;
+            }
+            current = current.next;
+        }
+        return null;
+    }
+
+    private int getIndex(K key, int capacity) {
+        return key == null ? 0 : Math.abs(key.hashCode()) % capacity;
+    }
+
+    private void addToLink(Node<K,V> newItem, int index, Node<K, V>[] table) {
+        Node<K, V> current = table[index];
+        while (current != null) {
+            if (current.key == newItem.key
+                    || current.key != null && current.key.equals(newItem.key)) {
+                current.value = newItem.value;
+                size--;
+                return;
+            }
+            if (current.next == null) {
+                current.next = newItem;
+                return;
+            }
+            current = current.next;
+        }
+    }
+
+    private Node<K, V>[] resize() {
+        int newCap = table.length * 2;
+        Node<K, V>[] newTab = (Node<K,V>[])new Node[newCap];
+        transfer(newTab);
+        return newTab;
     }
+
+    private void transfer(Node<K, V>[] table) {
+        for (Node<K, V> kvNode : this.table) {
+            if (kvNode != null) {
+                if (kvNode.next != null) {
+                    Node<K, V> current = kvNode;
+                    while (current != null) {
+                        putToNewTable(current, table);
+                        current = current.next;
+                    }
+                    continue;
+                }
+                putToNewTable(kvNode, table);
+            }
+        }
+    }
+
+    private void putToNewTable(Node<K,V> node, Node<K,V>[] table) {
+        if (node.key == null) {
+            if (table[0] != null) {
+                addToLink(new Node<K, V>(null, node.value), 0, table);
+            } else {
+                table[0] = new Node<K, V>(null, node.value);
+            }
+            return;
+        }
+        Node<K, V> newItem = new Node<>(node.key, node.value);
```

Comment:
```suggestion
    private void putToNewTable(Node<K,V> node, Node<K,V>[] table) {
        Node<K, V> newItem = new Node<>(node.key, node.value);
```
цей код загалом зайвий, нали хендляться нормально в перевірці і в getIndex()